src/core/data.jl

```diff
@@ -738,3 +738,86 @@ function _nan2zero(b, a; val=0)
     b[and.(isnan.(b), a.==val)] .= 0.0
     return b
 end
+
+
+"Counts active ungrounded connections on edge components"
+function count_active_connections(data::Dict{String,<:Any})
+    data_model = get(data, "data_model", MATHEMATICAL)
+    edge_elements = data_model == MATHEMATICAL ? PowerModelsDistribution._math_edge_elements : PowerModelsDistribution._eng_edge_elements
+    # bus_connections = Dict(id => [] for (id, _) in data["bus"])
+    active_connections = 0
+
+    for edge_type in edge_elements
+        for (_, component) in get(data, edge_type, Dict())
+            counted_connections = Set([])
+            if edge_type == "transformer" && !haskey(component, "f_connections") && data_model == ENGINEERING
+                for (wdg, connections) in enumerate(component["connections"])
+                    for terminal in connections
+                        if !(terminal in counted_connections)
+                            if !(terminal in data["bus"][component["bus"][wdg]]["grounded"])
+                                push!(counted_connections, terminal)
+                                active_connections += 1
+                            end
+                        end
+                    end
+                end
+            else
+                for (bus, connections) in [(component["f_bus"], component["f_connections"]), (component["t_bus"], component["t_connections"])]
+                    for (i, terminal) in enumerate(connections)
+                        if !(terminal in counted_connections)
+                            if data_model == ENGINEERING
+                                if edge_type == "transformer" && component["configuration"] == WYE && terminal != connections[end]
+                                    push!(counted_connections, terminal)
+                                    active_connections += 1
+                                elseif !(terminal in data["bus"][bus]["grounded"])
+                                    push!(counted_connections, terminal)
+                                    active_connections += 1
+                                end
+                            else
+                                if edge_type == "transformer" && component["configuration"] == WYE && terminal != connections[end]
+                                    push!(counted_connections, terminal)
+                                    active_connections += 1
+                                elseif !data["bus"]["$bus"]["grounded"][i]
+                                    push!(counted_connections, terminal)
+                                    active_connections += 1
+                                end
+                            end
+                        end
+                    end
+                end
+            end
+        end
+    end
+    return active_connections
+end
+
+
+"Counts active ungrounded terminals on buses"
+function count_active_terminals(data::Dict{String,<:Any}; count_grounded::Bool=false)
```

Comment:
Same concern as for the connections, though since you allow to count grounded terminals, and do not consider to state of the buses, now I really don't know what 'active' refers to.

Reply:
suggestion for a better name?